File: scripts/add_collapsible.py
import os, re

def update_toolkit(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    lines = content.split('\n')
    # Find sectionTitle function
    title_def_start = -1
    title_def_end = -1
    for i in range(len(lines)):
        line = lines[i].strip()
        if 'sectionTitle' in line and '(' in line and '=>' in line:
            title_def_start = i
            break
    if title_def_start == -1:
        print(f'  WARNING: no sectionTitle found in {filepath}')
        return
    title_def_end = lines[title_def_start + 1:]
    for i in range(title_def_start, len(lines)):
        if ')' in lines[i] and ')' in lines[i]:
            title_def_end = i
            break
    print(f'  {os.path.basename(filepath)}: sectionTitle at lines {title_def_start + 1}-{title_def_end}')
    func_body = lines[title_def_start + 1:title_def_end]
    func_def = '\n' + '\n'.join(func_body) + '\n'
    new_func = '\n' + func_def.replace(
        'const sectionTitle = (text: string) => (',
        'const sectionTitle = (text: string, sectionId: string) => (\n' +
    )
  )
    content = content[:title_def_start] + new_func + content[title_def_end:]
    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(content)
        print(f'  Updated {os.path.basename(filepath)}')

def main():
    toolkit_dir = '/home/z/my-project/src/components/room/widgets'
    for fname in os.listdir(toolkit_dir):
        if fname.endswith('Toolkit.tsx'):
            update_toolkit(os.path.join(toolkit_dir, fname))
if __name__ == '__main__':
    main()

File: scripts/add_collapsible.mjs
const fs = require('fs');
module.exports = {};

function updateToolkit(filepath) {
  let content = fs.readFileSync(filepath, 'utf-8');
  const lines = content.split('\n');
  let titleDefStart = -1, titleDefEnd = -1;
  for (let i = 0; i < lines.length; i++) {
    const line = lines[i];
    if (line.includes('sectionTitle') && line.includes('(')) {
      titleDefStart = i; break;
    }
  }
  if (titleDefStart === -1) { return; }
  titleDefEnd = titleDefStart + 1;
  for (let i = titleDefStart; i <= titleDefEnd; i++) {
    if (')' === lines[i].trim()) {
      titleDefEnd = i + 1; break;
    }
  }
  const bodyLines = lines.slice(titleDefEnd);
  const chevronHtml = '<span className="section-chevron">&#x2212;</span>';
  const newBody = bodyLines.map(l => l.trim()).join('\n').replace('className={', 'className={').replace('</span>', '</span>'));
  const newFunc = '\n' + lines[titleDefStart] + newBody + '\n';
  fs.writeFileSync(filepath, newFunc, 'utf-8');
  console.log('Updated ' + filepath);
}
module.exports.updateToolkit = updateToolkit;
